### DISCUSSION TITLE
iNav 2.2.1 and Walksnail HD video (OSD)

### DISCUSSION THREAD

**Discussion by mazi14** [iNavFlight/inav#10304] [Q&A] [answered]
I have got a plane with Matek 411 FC and iNav 2..2.1 flashed in. I would like to step into digital video transmission now and ordered Walksnail HD kit. Is this FW(2.2) capable of "Display port" to see OSD in Avatar X goggles (how to set it up in iNav?) or I have to update iNav to version 6 at least?

**Comment by mmosca** [comment, marked as answer]
Upgrade to INAV 7.1.2.

Digital video didn't exist when INAV 2.2.1 was released.

**Reply by mazi14** [reply]
> Upgrade to INAV 7.1.2.
> 
> Digital video didn't exist when INAV 2.2.1 was released.

Thank you for your quick reply. I will update to recommended  (latest) 7.1 version, but which first release was capable to manage it? I am a lttle bit nervous to skip so many versions. I will have to set up my FC from the scratch, will I?

**Reply by b14ckyy** [reply]
First version that had full Avatar support was 6.0 with updates later down the line. So you have to set up from scratch anyway. 
Go to 7.1.2 and you are good. Will be the last official version for F411 FCs anyway.

**Reply by mazi14** [reply]
Finally, I bought new FC like you adviced. But there is one thing, which I noticed after I upgraded to iNav 7.1. When I pull rudder stick (mode 1) fully to the right, it overrides pre-flight checks. Is it new default behavioral? Because I am not aware, that I would set up such a function.

**Reply by MrD-RC** [reply]
> Finally, I bought new FC like you adviced. But there is one thing, which I noticed after I upgraded to iNav 7.1. When I pull rudder stick (mode 1) fully to the right, it overrides pre-flight checks. Is it new default behavioral? Because I am not aware, that I would set up such a function.

This is normal.

**Comment by mazi14** [comment]
Great support, thanks! I have two more questions related to Walksnail and iNav. Goggles manual says, that video recording (I guess onboard too) can be triggered by FC ("takeoff REC" option) by arm switch. Is any iNav set up needed for this functionality? And is it possible to set VTx power level via FC like by smartaudio for analog system?

**Reply by b14ckyy** [reply]
No setup needed. The VTX will get the arming state through the same serial connection that delivers OSD data. This just works.

**Reply by mazi14** [reply]
I updated my FC to iNav 7.1.2 and OSD works in Avatar goggless, but I found out HW limitation of my FC Matek F411 wing. The problem is, that this FC has got 2 UARTs only. Until now (for analog video transmittion) I used UART1 Tx for FPort (FrSky receiver with telemetry), UART2 for GPS and SoftSerial for SmartAudio VTx control. Now, for digital video, I am missing one UART for MSA DisplayPort as there is a warning in iNav configurator, that SoftSerial port should not been used for flight critical devices such a GPS, receiver or DisplayPort, Do you have any advice, how to solve this problem?
![iNav softserial caution](https://github.com/user-attachments/assets/fb5bb204-abb2-4a21-b21e-f45b9cb28b70)

**Reply by b14ckyy** [reply]
Well, that's the main reason why F411 won't get further official INAV updates. WIth 2 UART they are not up to the task anymore. 
But as Walksnail OSD seems to not work at all over softserial, I suggest to put the GPS on softserial and the Walksnail on UART1 (RC link should always be on UART2 or 4 or higher. Don't put RC link on UART1 or 3 as it can cause issues to get the FC updated in DFU mode. 

keep an eye on your GPS in flight then to make sure its stable before you rely on any navigation mode.

**Reply by mazi14** [reply]
I have difficulty to get GPS running on Softserial. I flashed FD SFTSRL FW into F411 FC, connected GPS Tx to LED pin of the FC and GPS Rx to ST1, but GPS does not send any messages. Am I missing something or am I doing something wrong? I tried to swap Tx and Rx wires, but this did not help.

**Reply by b14ckyy** [reply]
some GPS just don't work on Softserial at all. Like some BN220 on a F411-WSE. No chance then. There is not much we can do in that case. Only way is to either keep it analog or upgrade the FC to something with 3 or more UART.

**Reply by mazi14** [reply]
Thank you for hint. It is my case - GPS BN 220. Are there any GPS sensors, that work on Softserial then? Which ones? Swaping GPS seems to me like the cheapest solution at the moment.

**Reply by stronnag** [reply]
> Thank you for hint. It is my case - GPS BN 220. Are there any GPS sensors, that work on Softserial then? Which ones? Swaping GPS seems to me like the cheapest solution at the moment.

Unlikely, softserial is inherently flawed for high intensity I/O. Just get a better FC and be done.

**Reply by mazi14** [reply]
OK. Could anybody clarify to me, why GPS and DisplayPort need duplex comunication with FC? Until now I thought, that GPS sensor just sends data to FC and DisplayPort just receives them from FC.

**Reply by mmosca** [reply]
It is not about duplex or not. Softserial is just not good for higher speeds needed for gps and displayport.

INAV needs to bi-directional comms with the GPS so ti can setup what messages it needs to receiver and what satelites to look for, among other things.

Digital VTXs also need to ask information from the flight controller, to identify if it the fc is armed so it can go into full power and start recording or if it is INAV, ArduPilot or BF so it can use the correct font.

**Comment by mmosca** [comment]
Yes.

https://www.mrd-rc.com/tutorials-tools-and-testing/inav-flight/inav-stick-commands-for-all-transmitter-modes/
